pysph/sph/scheme.py

```diff
@@ -97,7 +97,8 @@ def setup_properties(self, particles, clean=True):
 
     # Private protocol ###################################################
 
-    def _ensure_properties(self, pa, desired_props, clean=True):
+    def _ensure_properties(self, pa, desired_props, stride_info=None,
```

Comment:
put the stride_info at the end, or you may have to track all uses of `_ensure_properties` where clean is set using positional arguments.